SPEC.md

```diff
@@ -211,6 +211,65 @@ Plugins may define additional fields that they accept and may generate an error
 }
 ```
 
+### Network Configuration Lists
+
+Network configuration lists provide a mechanism to run multiple CNI plugins for a single container in a defined order, passing the result of each plugin to the next plugin.
+The list is composed of well-known fields and list of one or more standard CNI network configurations (see above).
+
+The list is described in JSON form, and can be stored on disk or generated from other sources by the container runtime. The following fields are well-known and have the following meaning:
+- `cniVersion` (string): [Semantic Version 2.0](http://semver.org) of CNI specification to which this configuration list and all the individual configurations conform.
+- `name` (string): Network name. This should be unique across all containers on the host (or other administrative domain).
+- `plugins` (list): A list of standard CNI network configuration dictionaries (see above).
+
+When executing a plugin list, the runtime MUST replace the `name` and `cniVersion` fields in each individual network configuration in the list with the `name` and `cniVersion` field of the list itself.
+This ensures that the name and CNI version is the same for all plugin executions in the list, preventing versioning conflicts between plugins.
+
+For the ADD action, the runtime MUST also add a `prevResult` field to the configuration JSON of any plugin after the first one, which MUST be the Result of the previous plugin (if any) in JSON format.
```

Comment:
@asridharan to keep things simplest, I would advocate two things:

1) plugins should not return errors on non-critical failures; eg if some resources they want aren't present (like the container netns) they shouldn't fail.  Absolutely critical failures can still be returned.
2) the runtime calls *all* plugins for DEL, regardless of which ones were ADD-ed.  Since plugins should be flexible on DEL, this shouldn't fail even if the plugin was not run.

This approach means the runtime doesn't have to store a specific list of plugins, and also means we don't have to do complex explanations in the spec that could be misunderstood.  I'm a big fan of making things simple and unambiguous rather than potentially convoluted.

Plus I think plugins should be resilient on DEL anyway...